### PR TITLE
fix/filters

### PR DESCRIPTION
**Pull request by decanus**
This pull request changes the way message filters work, as discussed with @oskarth notify is now being called in the `rpcHandler` function. Additionally the filters now receive a `Message` type.

### CODE REVIEW

File: waku/protocol/v2/filter.nim [hunk 1]
```diff
@@ -1,9 +1,11 @@
+import libp2p/protocols/pubsub/rpc/messages
+
 import 
   tables
 
 type 
 
-  FilterMessageHandler* = proc(msg: seq[byte]) {.gcsafe, closure.}
+  FilterMessageHandler* = proc(msg: Message) {.gcsafe, closure.}
```
Comment (oskarth) [inline waku/protocol/v2/filter.nim:8]: how come this isn't Topichandler or ContentHandler per https://github.com/status-im/nim-waku/blob/master/docs/api/v2/node.md? What is Filter and notify and do we need to add new names?

File: waku/protocol/v2/waku_protocol2.nim [hunk 2]
```diff
@@ -117,6 +116,11 @@ method rpcHandler*(w: WakuSub,
                    rpcMsgs: seq[RPCMsg]) {.async.} =
   debug "rpcHandler"
 
+  for rpcs in rpcMsgs:
```
Comment (oskarth) [inline waku/protocol/v2/waku_protocol2.nim:119]: might want to have this after procCall/super like other places in code base

File: waku/protocol/v2/filter.nim (hunk 1 above)
Comment (decanus) [reply waku/protocol/v2/filter.nim:8]: @oskarth, we could name it that, but filters are more for internode communication. Like various components within the node, so I am not sure if those match.

Comment (oskarth) [reply waku/protocol/v2/filter.nim:8]: What is internode communication? Either it is part of the protocol, the node API or it is an implementation detail to achieve one or the other.

Comment (decanus) [reply waku/protocol/v2/filter.nim:8]: its an implementation detail to be able to save messages recieved to the mailserver. Not sure how you'd want to put that

Comment (oskarth) [reply waku/protocol/v2/filter.nim:8]: There is no mailserver in Waku v2 but fine, if implementation detail lets merge for now

**Pull request merged**